src/internal/components/token-list/index.tsx

```diff
@@ -77,7 +78,14 @@ export default function TokenList<Item>({
   return (
     <div className={clsx(styles.root, styles.vertical)}>
       {hasVisibleItems && (
-        <ul id={controlId} className={clsx(styles.list, styles[alignment])}>
+        <ul
+          id={controlId}
+          className={clsx(styles.list, {
+            [styles.vertical]: alignment === 'vertical',
+            [styles.horizontal]: alignment === 'horizontal' && !isGrid,
+            [styles.grid]: alignment === 'horizontal' && isGrid,
```

Comment:
Should we instead add a new alignment type: "horizontal-grid" to clearly indicate there is no vertical grid?